gammapy/makers/utils.py

```diff
@@ -422,3 +424,161 @@ def make_theta_squared_table(
     table.meta["ON_RA"] = position.icrs.ra
     table.meta["ON_DEC"] = position.icrs.dec
     return table
+
+
+def get_rad_max_vs_energy(rad_max, pointing, geom):
+    """Obtain the values of `RAD_MAX` at a given offset and for an array of
+    estimated energy values (in the geom energy axis).
+
+    Parameters
+    ----------
+    rad_max : `~gammapy.irf.RadMax2D`
+        the RAD_MAX_2D table IRF
+    geom : `~gammapy.maps.Geom`
+        the map geom to be used
+    pointing : `~astropy.coordinates.SkyCoord`
+        pointing direction
+
+    Returns
+    -------
+    array : `~astropy.units.Quantity`
+        Values of the `RAD_MAX` corresponding to each estimated energy bin center.
+    """
+    on_center = geom.center_skydir
+    offset = on_center.separation(pointing)
+
+    rad_max_vals = rad_max.evaluate(
+        offset=offset, energy=geom.axes["energy"].center
+    )
+
+    return rad_max_vals
+
+
+def make_counts_rad_max(geom, rad_max, events):
+    """Extract the counts using for the ON region size the values in the
+    `RAD_MAX_2D` table.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.RegionGeom`
+        reference map geom
+    rad_max : `~gammapy.irf.RadMax2D`
+        the RAD_MAX_2D table IRF
+    events : `~gammapy.data.EventList`
+        event list to be used to compute the ON counts
+
+    Returns
+    -------
+    counts : `~gammapy.maps.RegionNDMap`
+        Counts vs estimated energy extracted from the ON region.
+    """
+    rad_max = get_rad_max_vs_energy(rad_max, events.pointing_radec, geom)
```

Comment:
A question here. Is the rad_max determination usually done directly from the event energy or from the bin energy? i.e. is it expected that different energy binning yield different select event lists?

Reply:
Energy binning of what? I'd say you look up the cut value based on the event energy, like we do here: 

https://github.com/cta-observatory/pyirf/blob/3f9cfba672c8a25eb5094d25acf411c6b2104112/pyirf/cuts.py#L104